aws/resource_aws_iam_role_test.go

```diff
@@ -375,3 +394,71 @@ POLICY
 }
 `, rName)
 }
+
+func testAccAWSIAMRoleConfig_force_detach_policies(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_iam_role_policy" "test" {
```

Comment:
Technically we're not really testing the new functionality here IMO, because the policy will get destroyed first due to its relationship and `force_destroy` won't get chance to ever take action.
I think we'd need to remove those policies from state as part of the test in order to verify it actually works.

Reply:
That's right👍
But I couldn't have an idea to realize it... Could you give me advice?

Reply:
So I think the easier way is to add policies outside of the config, instead of removing them from state. We could add the following to your new test:

```
testAccAddAWSIAMRolePolicy(&conf),
```
which would just call the API to add a policy outside of any Terraform CRUD.

Testing framework will then attempt to destroy the role and exercise this new functionality.